astropy/modeling/functional_models.py

```diff
@@ -400,13 +400,15 @@ def evaluate(x, factor):
         return factor * x
 
 
-class Redshift(Fittable1DModel):
+class RedshiftWavelength(Fittable1DModel):
     """
     One dimensional redshift model.
 
+    Redshifts a wavelength array from rest (z=0) to the input redshift.
+
     Parameters
     ----------
-    z : float or a list of floats
+    z : int or float
```

Comment:
Looking elsewhere in this module ISTM that most parameters are documented to take a `float` which I think is fine for now since that's the most basic use case.  A more detailed reading of the docs can reveal that anything array-like is acceptable too.  In general (unless explicitly stated otherwise) any Python function that takes a `float` argument should be assumed to accept `int` as well, which will be cast to a float in most operations that would otherwise involved floats.